frontend/src/components/ui/StationListWindow/hooks/useInfiniteStationSummaries.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect } from 'react';
+
+import { useInfiniteQuery, useQueryClient } from '@tanstack/react-query';
+
+import { fetchStationSummaries } from '@hooks/fetch/fetchStationSummaries';
+
+import { cachedStationSummariesActions } from '@ui/StationListWindow/tools/cachedStationSummaries';
+
+import { QUERY_KEY_STATION_SUMMARIES } from '@constants/queryKeys';
+
+import type { StationMarker, StationSummary } from '@type';
+
+interface StationSummaryResponse {
+  stations: StationSummary[];
+  nextPage: number;
+}
+
+const makeStationIdsChunks = (markers: StationMarker[]) => {
+  return markers.reduce((acc: string[][], marker, index) => {
```

Comment:
바뀐 방식에서 useInfiniteQuery 훅을 활용하기 때문에 이제는 stationIdChunks를 만들 때 filter로 미리 캐시된 정보를 제거해도 될 것 같습니다.

이전 방식에서는 
1. 캐시된 정보를 보여준다.
2. 충전소 마커 요청을 통해 받아온 정보로 stationIdChunks를 만든다.
3. 리스트의 끝에 닿으면 stationIdChunks의 page 인덱스에 해당하는 충전소 아이디로 충전소 목록 조회 요청을 발생시킨다.
4. page 상태에 1을 더해준다.
5. 위 과정을 반복한다.

위와 같은 과정을 통해 리스트를 보여줬으므로 stationIdChunks에서 미리 캐시 데이터를 제외하게 되면 page는 올라갔는데 stationIdChunks의 길이는 줄어든 상태로 요청을 발생시키게 되어 page와 stationIdChunks의 인덱스의 불일치로 인해 화면 내 모든 마커의 목록을 완성할 수 없었습니다. 그래서 어쩔 수 없이 새로운 요청을 10개씩 발생시키는 것을 포기하고 요청을 발생시키는 그 순간에 캐시 데이터에 지금 요청을 발생시킬 id가 포함되어 있는지 여부를 판별했었는데 이 방식이 이제는 불필요해보입니다.

현재 방식에서는
1. 캐시된 정보를 보여준다.
2. useInfiniteQuery를 통해 캐시되지 않은 정보들을 요청하고, 화면에 출력한다.

위와 같이 방식이 변경되었으므로 page와 stationIdChunks의 인덱스 간의 불일치가 발생하지 않을 것이므로

```suggestion
  return markers
    .filter((marker) => !cachedStationSummariesActions.has(marker.stationId))
    .reduce((acc: string[][], marker, index) => {
```
위와 같이 stationIdChunks를 만들 때 캐시데이터를 미리 제거해도 괜찮을 것 같습니다!

Reply:
아 그렇네요.
생각해보니 이로 인해서 항상 일정한 데이터를 요청하는 것이 아닌 불균형하게 들쑥날쑥 요청이 들어가던 문제도 덩달아 해결되겠군요